coil-base/src/main/java/coil/request/DefaultRequestOptions.kt

```diff
@@ -18,8 +18,8 @@ import kotlinx.coroutines.Dispatchers
 class DefaultRequestOptions(
     val interceptorDispatcher: CoroutineDispatcher = Dispatchers.Main.immediate,
     val fetcherDispatcher: CoroutineDispatcher = Dispatchers.IO,
-    val decoderDispatcher: CoroutineDispatcher = Dispatchers.IO,
-    val transformationDispatcher: CoroutineDispatcher = Dispatchers.IO,
+    val decoderDispatcher: CoroutineDispatcher = Dispatchers.Default,
+    val transformationDispatcher: CoroutineDispatcher = Dispatchers.Default,
```

Comment:
Yeah, you're right. I observed lagging in demo when switch from MP4 screen to JPG, I thought 100% CPU is stolen, but I ran CPU profiler and saw the CPU isn't stolen that much, it just because `Dispatchers.IO` is out of room (`MediaMetadataRetriever` hanging), and the `ImageListAdapter`'s differ is also scheduled in `Dispatchers.IO`.
https://github.com/coil-kt/coil/blob/986e91ddb58ac5a4f55ccee99140a0ed155d8e9a/coil-sample/src/main/java/coil/sample/Utils.kt#L53-L57

Reply:
> MediaMetadataRetriever being slow and not supporting cancellation

Did you try `runInterruptible` or just call `MediaMetadataRetriever#realease()` early to support cancellation?

Reply:
@Khang-NT Sorry forgot to respond to this. I tried out `runInterruptible` and calling `release` if the coroutine is cancelled and unfortunately neither free up the thread that's decoding. In fact, I was able to get `MediaMetadataRetriever` to completely lock up and never return by using `release` while the decode was in progress.